scopegen/generate/go.go

```diff
@@ -20,11 +21,44 @@ const (
 )
 
 {{ end -}}
-var AuthScopes = map[string][]AuthScope{
+var authScopes = map[string][]AuthScope{
 {{- range $method := .Methods }}
 	"{{ $method.Name }}": []AuthScope{ {{- range $index, $scope := $method.Scopes }}{{ if $index }}, {{ end }}Scope_{{ $scope }}{{ end -}} },
 {{- end }}
 }
+
+// Any allows a loose challenge, for claims containing any of the method scopes.
+//
+// Use All method as a defualt for OAuth2 style scopes.  Any is usful with more complex scope definitions.
+func (svcSc ServiceScope) Any(method string, claims []string) bool {
```

Comment:
maybe just make Any/All a global function? I don't see why we need ServiceScope struct.

Reply:
The ServiceScope struct allows it to be passed as an interface.  All services would be able to define their scopes using `<svc>.ServiceScope` as an embedded field.  Interceptors can use that interface via type assertion, with minimal changes to the service implementation itself.

Reply:
Could you add an code example somewhere in README.md for this?

Reply:
I plan on creating a basic interceptor, using that to show its use.